src/sinks/tcp.rs

```diff
@@ -50,7 +76,48 @@ impl SinkConfig for TcpSinkConfig {
             .next()
             .ok_or_else(|| "Unable to resolve DNS for provided address".to_string())?;
 
-        let sink = raw_tcp(addr, acker, self.encoding.clone());
+        let tls = match self.tls {
+            Some(ref tls) => {
+                if tls.enabled.unwrap_or(false) {
+                    if tls.key_file.is_some() != tls.crt_file.is_some() {
+                        return Err("Must specify both tls key_file and crt_file".into());
+                    }
+                    let add_ca = match tls.ca_file {
+                        None => None,
+                        Some(ref filename) => Some(load_certificate(filename)?),
+                    };
+                    let identity = match tls.crt_file {
+                        None => None,
+                        Some(ref filename) => {
```

Comment:
btw you can drop the `ref` here for borrowing in the match

`match &tls.crt_file`

Reply:
Done